File: BookStore-SPA/src/app/_models/Book.ts
export class Book {
  id: number;
  name: string;
  author: string;
  description: string;
  price: number | null;
  publishDate: any;
  categoryId: number | null;
  pieces: number | null;

  constructor(
    id: number,
    name: string,
    author: string,
    description: string,
    price: number | null,
    publishDate: any,
    categoryId: number | null,
    pieces: number | null
  ) {
    this.id = id;
    this.name = name;
    this.author = author;
    this.description = description;
    this.price = price;
    this.publishDate = publishDate;
    this.categoryId = categoryId;
    this.pieces = pieces;
  }
}

File: Bookstore.Domain/Models/Book.cs
using System;

namespace BookStore.Domain.Models
{
    public class Book : Entity
    {
        public string Name {get;set;}
        public string Author { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public DateTime PublishDate { get; set; }
        public int CategoryId { get; set; }
        public int Pieces { get; set; }
        /* EF Relation */
        public Category Category { get; set; }

        public string FileName { get; set; }
    }
}